bindings/sundials4py/cvodes/generate.yaml

```diff
@@ -0,0 +1,72 @@
+# -----------------------------------------------------------------
+# Programmer(s): Cody J. Balos @ LLNL
+# -----------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2002-2025, Lawrence Livermore National Security
+# and Southern Methodist University.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# -----------------------------------------------------------------
+# This YAML file is used to instruct the generate.py script on
+# how to generate some of the nanobind code needed for the
+# CVODES module of sundials4py.
+# -----------------------------------------------------------------
+
+modules:
+  all:
+    fn_exclude_by_name__regex:
+    - "Free" # Free and destroy functions should not need to be called as all objects on the Python side are RAII
+    - "Destroy"
+    - "Space" # Space functions are deprecated, so dont expose them in Python
+    # Due to the need to convert between sys.argv and C argv, we need to do custom wrappers of these
+    - "SetOptions" 
+    macro_define_include_by_name__regex:
+    - "^SUN_"
+    - "^CV_"
+    - "^CVLS_"
+  cvodes:
+    path: cvodes/cvodes_generated.hpp
+    headers:
+    - ../../include/cvodes/cvodes.h
+    - ../../include/cvodes/cvodes_ls.h
+    - ../../include/cvodes/cvodes_proj.h
+    # this option describes the functions which have optional pointer arguments,
+    # i.e., one where you could provide NULL
+    fn_params_optional_with_default_null:
+      "SetLinearSolver":
+      - "A" 
+    fn_exclude_by_name__regex:
+    # We do custom handling of Create so we can wrap the void* in a CVodeView
+    - "^CVodeCreate$"
+    # we use user_data for sneaking in python contexts, so we don't interface these
+    - "^CVodeGetUserData$" 
+    - "^CVodeSetUserData$" 
+    - "^CVodeGetUserDataB$" 
+    - "^CVodeSetUserDataB$"
+    # this function should be deprecated, so we don't interface it
+    - "^CVodeSetMonitorFn$"
```

Comment:
I don't think we've deprecated the Monitor functions yet, but I agree we should do that it in this release for removal in v8.